packages/shared/src/hooks/useTagAndSource.ts

```diff
@@ -1,36 +1,48 @@
 import { useContext } from 'react';
+import { Features, isFeaturedEnabled } from '../lib/featureManagement';
 import AuthContext from '../contexts/AuthContext';
 import AnalyticsContext from '../contexts/AnalyticsContext';
 import useMutateFilters from './useMutateFilters';
 import { Source } from '../graphql/sources';
 import AlertContext from '../contexts/AlertContext';
+import FeaturesContext from '../contexts/FeaturesContext';
 
 export interface TagActionArguments {
   tags: Array<string>;
   category?: string;
+  requireLogin?: boolean;
 }
 
 export interface SourceActionArguments {
   source: Source;
+  requireLogin?: boolean;
+}
+
+interface UseTagAndSourceProps {
+  origin?: string;
+  postId?: string;
+}
+
+interface UseTagAndSource {
+  onFollowTags: (params: TagActionArguments) => void;
+  onUnfollowTags: (params: TagActionArguments) => void;
+  onBlockTags: (params: TagActionArguments) => Promise<boolean>;
+  onUnblockTags: (params: TagActionArguments) => Promise<unknown>;
+  onFollowSource: (params: SourceActionArguments) => Promise<unknown>;
+  onUnfollowSource: (params: SourceActionArguments) => Promise<unknown>;
 }
 
 export default function useTagAndSource({
   origin,
   postId,
-}: {
-  origin?: string;
-  postId?: string;
-}): {
-  onFollowTags: ({ tags, category }: TagActionArguments) => void;
-  onUnfollowTags: ({ tags, category }: TagActionArguments) => void;
-  onBlockTags: ({ tags }: TagActionArguments) => Promise<unknown>;
-  onUnblockTags: ({ tags }: TagActionArguments) => Promise<unknown>;
-  onFollowSource: ({ source }: SourceActionArguments) => Promise<unknown>;
-  onUnfollowSource: ({ source }: SourceActionArguments) => Promise<unknown>;
-} {
+}: UseTagAndSourceProps): UseTagAndSource {
+  const { flags } = useContext(FeaturesContext);
+  const shouldShowMyFeed = isFeaturedEnabled(Features.MyFeedOn, flags);
   const { alerts, updateAlerts } = useContext(AlertContext);
   const { user, showLogin } = useContext(AuthContext);
   const { trackEvent } = useContext(AnalyticsContext);
+  const shouldShowLogin = (requireLogin?: boolean) =>
+    (!shouldShowMyFeed && !user) || (requireLogin && !user);
```

Comment:
I think this is a wrong condition.
Because we require login for following tags but if you are part of the my feed experiment you shouldn't be asked to login. But the condition will still yield true.

Reply:
When blocking sources and tags through the post menu options, I was thinking it should require login since they might think it should've been blocked but didn't happen.

So being part of the experiment can only happen in the actual FeedFilters in the sidebar.

Reply:
Let me refactor this for better readability.